host-bmc/dbus/deserialize.cpp

```diff
@@ -0,0 +1,232 @@
+         pldm::dbus::CustomDBus::getCustomDBus().implementGlobalInterface(path);
+     }},
+    {"LicenseEntry", [](const std::string& path, Properties values) {
+         std::string name{};
+         std::string serialno{};
+         dbus::LicenseEntryType type;
+         dbus::LicenseEntryAuthorizationType authtype;
+         uint32_t authdevno{};
+         uint64_t exptime{};
+
+         if (values.contains("name"))
+         {
+             name = std::get<std::string>(values.at("name"));
+         }
+         if (values.contains("serialNumber"))
+         {
+             serialno = std::get<std::string>(values.at("serialNumber"));
+         }
+         if (values.contains("type"))
+         {
+             type = std::get<dbus::LicenseEntryType>(values.at("type"));
+         }
+         if (values.contains("authorizationType"))
+         {
+             authtype = std::get<dbus::LicenseEntryAuthorizationType>(
+                 values.at("authorizationType"));
+         }
+         if (values.contains("expirationTime"))
+         {
+             exptime = std::get<uint64_t>(values.at("expirationTime"));
+         }
+         if (values.contains("authDeviceNumber"))
+         {
+             authdevno = std::get<uint32_t>(values.at("authDeviceNumber"));
+         }
+
+         pldm::dbus::CustomDBus::getCustomDBus().implementLicInterfaces(
+             path, authdevno, name, serialno, exptime, type, authtype);
+     }}};
+
+std::set<uint16_t> getEntityTypes(const fs::path& path)
+{
+    std::set<uint16_t> entityTypes{};
+
+    if (!fs::exists(path) || fs::is_empty(path))
+    {
+        std::cerr << "The file does not exist or is empty, FILE_PATH = " << path
+                  << std::endl;
+        return entityTypes;
+    }
+
+    try
+    {
+        std::ifstream jsonFile(path);
+        auto json = Json::parse(jsonFile);
+
+        // define the default JSON as empty
+        const std::set<uint16_t> empty{};
+        auto objectPaths = json.value("entityTypes", empty);
+        std::ranges::transform(objectPaths,
+                               std::inserter(entityTypes, entityTypes.begin()),
+                               [](const auto& i) { return i; });
+    }
+    catch (const std::exception& e)
+    {
+        std::cerr << "Failed to parse config file, FILE_PATH = " << path
+                  << ", ERROR = " << e.what() << std::endl;
+    }
+
+    return entityTypes;
+}
+
+void restoreDbusObj(std::shared_ptr<HostPDRHandler> hostPDRHandler)
+{
+    if (!pldm::serialize::Serialize::getSerialize().deserialize())
+    {
+        return;
+    }
+
+    auto entityTypes = getEntityTypes(DBUS_JSON_FILE);
+    auto savedObjs = pldm::serialize::Serialize::getSerialize().getSavedObjs();
+
+    for (auto& [type, objs] : savedObjs)
+    {
+        if (!entityTypes.contains(type))
+        {
+            std::cerr << "The type is not in entityTypes, type = " << type
+                      << std::endl;
```

Comment:
Instead of this, change this to
```bash
if(!entityType.contains(type))
{
continue;
}
std::cout << "Restoring dbus of type : " << type << std::endl;```

Reply:
> Instead of this, change this to
> 
> ```shell
> if(!entityType.contains(type))
> {
> continue;
> }
> std::cout << "Restoring dbus of type : " << type << std::endl;```
> ```

Done